### ISSUE TITLE
NoMethodError for "attribute.strip!" with v3.1.5

### ISSUE THREAD

**Issue by ledermann** [rack/rack#2225]
After updating a Rails application to Rack `v3.1.5`, I get exceptions in https://github.com/rack/rack/blob/v3.1.5/lib/rack/request.rb#L647 when [Updatetime robot](https://uptimerobot.com) comes by:

```
NoMethodError: undefined method `strip!' for nil attribute.strip! 
```

![Honeybadger report](https://github.com/rack/rack/assets/3557/8d441c82-1392-4d13-ab12-6f1ee7eb9888)

Uptime Robots does a GET request with these headers:

```
{
  "User-Agent": "Mozilla/5.0+(compatible; UptimeRobot/2.0; http://www.uptimerobot.com/)",
  "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
  "Accept-Language": "en-US,en;q=0.8",
  "Connection": "close",
  "cache-control": "no-cache",
  "Referer": "https://my.example.com",
  "Accept-encoding": ""
}
```

It seems a Safe Navigation Operator (&.) is missing.

Belongs to https://github.com/rack/rack/security/advisories/GHSA-cj83-2ww7-mvq7
References https://github.com/rack/rack/commit/97952ac5ea99976b75469a19191ee71bde0ebb0f

**Comment by ioquatix** [comment]
Damn, thanks for the report, let me take a look.

**Comment by ioquatix** [comment]
@ledermann can you please tell me what version of Ruby you are running on, and what server too?

**Comment by Earlopain** [comment]
No header from the reproducer crashes but passing `", "` to `parse_http_accept_header` does

**Comment by ledermann** [comment]
I'm using Ruby 3.3.3, Puma 6.4.2, Rails 7.1.3.4.

**Comment by ioquatix** [comment]
Please check the linked PR.

**Comment by ledermann** [comment]
To complete my report, the invalid header field sent by @uptimerobot that causes the exception is as follows:

```
"HTTP_ACCEPT_ENCODING" => ", gzip"
```

So, your fix in #2226 handles this nicely.

**Comment by ioquatix** [comment]
I'm curious how you ended up with such a header?

**Comment by ledermann** [comment]
@ioquatix My Rails app is monitored by UptimeRobot, which sends a request to my app every 5 minutes. I have no control over the exact request. After updating my app to Rack 3.1.5, Honeybadger was notified about the exception, so the request headers can be inspected there. 
Using the app with a browser is not affected.

Despite the fix, the header is of course incorrect, so I have contacted @uptimerobot.

**Comment by ioquatix** [comment]
This should be fixed in v3.1.6 - are you able to test it out and report back? Thanks.

**Comment by ledermann** [comment]
Works fine, there are no exceptions after updating to `v3.1.6` 👍
